python/mspasspy/db/schema.py

```diff
@@ -597,5 +597,83 @@ def set_writeable(self, key):
             raise MsPASSError(key + ' is not defined', 'Invalid')
         self._main_dic[key]['readonly'] = False
 
+
+'''
+check if a mspass.yaml file is valid or not
+'''
+def is_basic_type(s):
+    if s in ["ObjectID","int","float","double","complex","bool","string","list","dict","bytes"]:
```

Comment:
Check out the `type` method:
https://github.com/wangyinz/mspass/blob/1020dd621b3cb9970858b0522b507767601325d0/python/mspasspy/db/schema.py#L238-L264
You can see that we actually supports more types (except for the "complex"). Note that I also used the `strip()` and `casefold()` there so that extra spaces and cases will also be automatically converted. You might want to do the same here.

Reply:
That's a good pre-checking step, which I didn't notice!

Reply:
Also, should we support the complex type here? I guess we should be consistent with the above code.

Reply:
Complex is not a BSON-supported type, so we cannot have it anyway.